web/src/ferrostar-map.ts

```diff
@@ -221,54 +215,6 @@ export class FerrostarMap extends LitElement {
     }
   }
 
-  firstUpdated() {
-    // Skip initialization of the map if the user has supplied one already via a slot!
-    const slotChildren =
-      this.shadowRoot!.querySelector("slot")?.assignedElements() || [];
-    if (slotChildren.length == 0 && this.map === null) {
-      this.initMap();
-    }
-  }
-
-  /**
-   * Initialize the MapLibre Map control.
-   *
-   * This is run by default on firstUpdated,
-   * but is skipped if the user adds a map of their own.
-   */
-  initMap() {
-    this.map = new maplibregl.Map({
-      container: this.shadowRoot!.getElementById("map")!,
-      style: this.styleUrl
-        ? this.styleUrl
-        : "https://demotiles.maplibre.org/style.json",
-      center: this.center ?? [0, 0],
-      pitch: this.pitch,
-      bearing: 0,
-      zoom: this.zoom,
-      attributionControl: { compact: true },
-    });
-
-    this.geolocateControl = new GeolocateControl({
-      positionOptions: {
-        enableHighAccuracy: true,
-      },
-      trackUserLocation: true,
-    });
-
-    this.map.addControl(this.geolocateControl);
-
-    this.map.on("load", (e) => {
-      if (this.geolocateOnLoad) {
-        this.geolocateControl?.trigger();
-      }
-
-      if (this.configureMap !== undefined) {
-        this.configureMap(e.target);
-      }
-    });
-  }
-
```

Comment:
I think this also assumes that the user must supply a component. I guess I don't necessarily have an issue with that, but as things stand today, there is no de facto standard npm package you can install that gives you a web component for MapLibre; you need to write your own (e.g. with Vue).

Reply:
I'm OK with this, generally, but there are a few things about the current implementation that are half-baked:

* The geolocateControl no longer exists. I think this is fine if we make it the user's responsibility to handle this, but we should clean up the code in this file. Similarly, `geolocateOnLoad` is currently broken and should be either fixed or removed.
* The `configureMap` property is also unused; either hook it up or delete it.
* Is the `maplibreglStyles` necessary here? I don't think so, since we don't actually load the map within the component anymore.